paperqa/llms.py

```diff
@@ -812,14 +833,24 @@ async def max_marginal_relevance_search(
             k: Number of results to return.
             fetch_k: Number of results to fetch from the vector store.
             embedding_model: model used to embed the query
+            partitioning_fn: optional function to partition the documents into
+                different groups, performing MMR within each group
```

Comment:
```suggestion
                different groups, performing MMR within each group.
```

https://peps.python.org/pep-0257/#one-line-docstrings
> The docstring is a phrase ending in a period.